api/v1/coupon.js

```diff
@@ -0,0 +1,86 @@
+// Dependencies
+var express = require('express');
+var fs = require('fs')
+var bodyParser = require('body-parser');
+
+// api router
+var router = express.Router();
+
+
+// ###### Coupons Block ######
+/* GET coupons list logic */
+router.get('/', function(req, res, next) {
+  res.render('coupon', { couponPage: 'Coupon' });
+  console.log('Get list of all coupons');
+});
+
+/* GET one coupon logic */
+router.get('/:couponID', function(req, res, next) {
+  var couponId = req.params.couponID;
+  console.log('Searching for one coupon by using couponID: ' + req.params);
+
+  // Use this path to get one coupons: /api/v1/coupons/123
+  // This can be used to send conpon json data back
+  if (req.params.couponID == 123) {
+  	res.send('Found Id: ' + couponId);
+  	console.log('Find ID: ' + couponId);
+  } else {
+  	res.send('Not Found Id: ' + couponId )
+  	console.log('Not Found ID ' + couponId);
+  }
+});
+
+/* POST one coupon logic */
+router.post('/', function(req, res, next) {
+  var receivedData = JSON.stringify(req.body);
```

Comment:
@LennyDuan In the app.js we have setup a body parser middleware ''' app.use(bodyParser.json()); ''' . so all request will be parsed into json object (if applicable) and can be accessed as req.body. 

so this line could be changed to var receivedData = req.body where receivedData would be a json object.